src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -411,5 +429,12 @@ void prepare_buffer_fusing::run(program& p) {
             node.get_output_layout();
             node.can_be_optimized(can_reshape_be_optimized(node));
         });
+        program_helpers::do_for_types<reduce>(*node, [&p](reduce_node& node) {
+            node.get_output_layout();
+            if (can_reduce_be_optimized(node)) {
+                node.can_be_optimized(true);
+                p.extract_and_remove(node);
+            }
+        });
```

Comment:
IMO, this is not related to this pass. It would be best if we can "not create" it at first place. If it is difficult not to create it, next place would be to create another pass or remove redundant reorder.

Reply:
I created another pass for removing redundant primitives.